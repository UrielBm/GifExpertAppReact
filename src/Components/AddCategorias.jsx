import React, {useState } from 'react'
import PropTypes from 'prop-types';
export const AddCategorias = ({setCategorias}) => {
    const [InputValue, setInputValue] = useState('');
    const OnChangeValue= (e) =>{
        setInputValue(e.target.value);
        console.log('Onchange llamado')
    };
    const handleSumit= (e)=>{
        e.preventDefault();
        if(InputValue.trim().length>2){
        setCategorias(categ=>[InputValue,...categ,]);
        setInputValue('')
        };
    };
    return (
        <form className="wrapper-form" onSubmit={handleSumit}>
            <p>{InputValue}</p>
            <label>
                <input type="text"  className="input-categoria" placeholder='"ingresa tu  nueva categoria"' value={InputValue} onChange={OnChangeValue} /> 
            </label>
        </form>
    );
}
AddCategorias.propTypes = {
    setCategorias: PropTypes.func.isRequired
};
