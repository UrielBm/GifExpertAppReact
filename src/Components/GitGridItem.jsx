import React from 'react'
import PropTypes from 'prop-types';
export const GitGridItem = ({id,title,url}) => {
    return (
        <div className="wrapper-gift-item">
            {/* {imagenes.title} */}
            < img  className="gift" src={url} alt={title} />
            <p className="titulo-gift animate__animated animate__bounceIn animate__delay-1s ">{title}</p>
        </div>
    )
}
GitGridItem.propTypes={
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
}