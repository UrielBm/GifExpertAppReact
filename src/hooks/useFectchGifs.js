import { getGifs } from "../helpers/getGifts";

const {useState, useEffect } = require("react");

 export const useFectchGifts=(categoria)=>{
     const [state, setstate] = useState({
         data: [],
         loading: true,
     });
     useEffect(()=>{
            getGifs(categoria)
            .then((imgs)=>{
                    // console.log(imgs);
                    setstate({
                        data: imgs,
                        loading: false,
                    })
            })
     },[categoria])
     return state; // {data:[], loading: true}
 }