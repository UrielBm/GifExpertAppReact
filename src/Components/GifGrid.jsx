import React, {Fragment}from 'react'
import { useFectchGifts } from '../hooks/useFectchGifs'
 import { GitGridItem } from './GitGridItem';
export const GifGrid = ({categoria}) => {
     const {loading,data:imagenes}=useFectchGifts(categoria);
    // console.log(data);

    return (
        <Fragment>
            <h3 className="titulo-Categoria">{categoria}</h3>
            {loading&& <p className="animate__animated animate__flash">Cargando...</p>}
            <div className="wrapper-card-gift-item animate__animated animate__bounceInDown">
                {
                    imagenes.map((imagenes)=>(
                        <GitGridItem key={imagenes.id} {...imagenes} />
                    ))
                }
                
            </div>
        </Fragment>
    )
}
