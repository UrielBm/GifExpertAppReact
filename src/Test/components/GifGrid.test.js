import React from 'react';
import '@testing-library/jest-dom';
import {shallow} from 'enzyme';
import { GifGrid } from '../../Components/GifGrid';
import { useFectchGifts } from '../../hooks/useFectchGifs';
jest.mock('../../hooks/useFectchGifs');
describe('Evaluando el componente GifGrid.jsx', () => {
    const categoria='Sherk';
    test('Deberia mostrar el componenete de manera correcta <GifGrind />  ', () => {
        useFectchGifts.mockReturnValue({
            data:[],
            loading: true,
        });
        const wrapper= shallow( <GifGrid categoria={categoria} />)
        expect(wrapper).toMatchSnapshot();
    });
    test('debe mostrar items cuando se carga imagenes useFetchGifs ', () => {
       const gift =[{
        id: 'ABC',
        url: 'https://localhost/cualquier/cosa.jpg',
        title:'Cualquier cosa',
       },
       {
        id: '123',
        url: 'https://localhost/cualquier2/cosa2.jpg',
        title:'Cualquier cosa2',
       },]
        useFectchGifts.mockReturnValue({
            data:gift,
            loading: false,
        });
        const wrapper= shallow( <GifGrid categoria={categoria} />)
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists() ).toBe(false);
        expect(wrapper.find('GitGridItem').length).toBe(gift.length);
    })
    
    
})
