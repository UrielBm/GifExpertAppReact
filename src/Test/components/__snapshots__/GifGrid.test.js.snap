// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Evaluando el componente GifGrid.jsx Deberia mostrar el componenete de manera correcta <GifGrind />   1`] = `
<Fragment>
  <h3
    className="titulo-Categoria"
  >
    Sherk
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Cargando...
  </p>
  <div
    className="wrapper-card-gift-item animate__animated animate__bounceInDown"
  />
</Fragment>
`;

exports[`Evaluando el componente GifGrid.jsx debe mostrar items cuando se carga imagenes useFetchGifs  1`] = `
<Fragment>
  <h3
    className="titulo-Categoria"
  >
    Sherk
  </h3>
  <div
    className="wrapper-card-gift-item animate__animated animate__bounceInDown"
  >
    <GitGridItem
      id="ABC"
      key="ABC"
      title="Cualquier cosa"
      url="https://localhost/cualquier/cosa.jpg"
    />
    <GitGridItem
      id="123"
      key="123"
      title="Cualquier cosa2"
      url="https://localhost/cualquier2/cosa2.jpg"
    />
  </div>
</Fragment>
`;
